### PR TITLE
Fixed change index pattern from menu doesnt work

### LINKED ISSUE #3186: Pivoting to index patterns besides wazuh-alerts-* broken in Wazuh web interface
| Wazuh | Elastic | Rev |
| ----- | ------- | --- |
| 4.1.4   | 7.10.0 with OpenDistro | 4105 |

Cannot pivot to other legitimate index patterns in web app, like `wazuh-archives-*`.  It always acts like it accepts the click but snaps back to `wazuh-alerts-*` every time.
In plain Discover mode I can move between the index patterns just fine.
This bug emerged sometime after Wazuh 4.0.4.

I've reproduced this on multiple Wazuh managersr and web browsers.

### PR DESCRIPTION
**Pull request by frankeros**
Hi team, this resolves
- Change the current index pattern from the Wazuh menu doesn't apply the selection.

Closes #3186
Related to #2985 and #3141

Testing: 
- Create a new index pattern (it should be by changing the default pattern setting, it should be set as default for Kibana too) 
- Change the pattern selection from the Wazuh menu to the new pattern.
- After refresh, the selection should be the new pattern.
- Change it again to another index pattern than the default.
- After refresh, the selection should be the pattern selected.
- The default pattern must be the same as previously configured.

### CODE REVIEW

**Review by Machi3mfl — APPROVED** [review]
Change index pattern from Wazuh menu FIXED.

File: public/services/resolves/get-ip.js [hunk 1]
```diff
@@ -30,12 +30,12 @@ export function getIp(
     const wazuhConfig = new WazuhConfig();
     const configuration = wazuhConfig.getConfig();
     const indexPatternFound = indexPatterns.find((indexPattern) => indexPattern.attributes.title === configuration.pattern);
-    if(!indexPatternFound){
-      AppState.removeCurrentPattern()
+    const currentPattern = AppState.getCurrentPattern();
+    if(!indexPatternFound && !indexPatterns.some((pattern) => pattern === currentPattern)){
+      AppState.removeCurrentPattern();
+      AppState.setCurrentPattern(indexPatternFound.id);
     }
```
Comment (Desvelao) [inline public/services/resolves/get-ip.js:37]: Test :x:
When change the index pattern using the Settings > Configuration section. After clink on the `Execute health check` buttons that appears when saving the configuration, the next screen is displayed:
image

@pablomarga suggests which the next line could be causing the problem
```
AppState.setCurrentPattern(indexPatternFound.id);
```

There 2 things that I am not understand:
- To get in to the conditional a condition is `indexPatternFound` variable is `falsy`. Why is using `indexPatternFound.id` here if `indexPatternFound` is not a object?
- The comparation of `!indexPatters.some((pattern => pattern === currentPattern))` is comparing a object `pattern` againts a posible `string`.

**Review by Desvelao — APPROVED** [review]
LGTM!

**Review by pablomarga — APPROVED** [review]
LGTM! Good job!

**Pull request merged**